dash-renderer/src/actions/setAppReadyState.js

```diff
@@ -0,0 +1,80 @@
+import {contains, filter, type} from 'ramda';
+import {createAction} from 'redux-actions';
+
+import Registry from './../registry';
+import {getAction} from './constants';
+
+export const SIMPLE_COMPONENT_TYPES = ['String', 'Number', 'Null', 'Boolean'];
+export const isSimpleComponent = component =>
+    contains(type(component), SIMPLE_COMPONENT_TYPES);
+
+const isAppReady = layout => {
+    const queue = [layout];
+
+    const res = {};
+
+    /* Would be much simpler if the Registry was aware of what it contained... */
```

Comment:
Would it? You still only want to wait for components in the layout...

Reply:
You're right, it wouldn't.